File: LeetCode75/Fibonacci_Number/Fibonacci_Number.cc
// 509 Fibonacci Number C++

class Solution {
public:
    int fib(int n) {
        if (n == 0) return 0;
        else if (n == 1) return 1;
        int pre = 1;
        int prepre = 0;
        for (int i = 1; i < n; i++) {
            int tmp = pre;
            pre += prepre;
            prepre = tmp;
        }
        return pre;
    }
};

File: LeetCode75_LevelOne/Fibonacci_Number/Fibonacci_Number.py
# 509 Fibonacci Number Python

class Solution(object):
    def fib(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n==0:
            return 0
        elif n==1:
            return 1
        pre = 1
        prepre = 0
        for i in range(n-1):
            tmp = pre
            pre = pre+prepre
            prepre = tmp
        return pre
